<!--
 Software License Agreement (BSD License)
 Copyright Â© 2022-2023 belongs to Shadow Robot Company Ltd.
 All rights reserved.
 Redistribution and use in source and binary forms, with or without modification,
 are permitted provided that the following conditions are met:
   1. Redistributions of source code must retain the above copyright notice,
      this list of conditions and the following disclaimer.
   2. Redistributions in binary form must reproduce the above copyright notice,
      this list of conditions and the following disclaimer in the documentation
      and/or other materials provided with the distribution.
   3. Neither the name of Shadow Robot Company Ltd nor the names of its contributors
      may be used to endorse or promote products derived from this software without
      specific prior written permission.
 This software is provided by Shadow Robot Company Ltd "as is" and any express
 or implied warranties, including, but not limited to, the implied warranties of
 merchantability and fitness for a particular purpose are disclaimed. In no event
 shall the copyright holder be liable for any direct, indirect, incidental, special,
 exemplary, or consequential damages (including, but not limited to, procurement of
 substitute goods or services; loss of use, data, or profits; or business interruption)
 however caused and on any theory of liability, whether in contract, strict liability,
 or tort (including negligence or otherwise) arising in any way out of the use of this
 software, even if advised of the possibility of such damage.
-->

<launch>
  <!-- By default we do not overwrite the URDF. Change the following to true to change the default behavior -->
  <arg name="load_robot_description" default="false"/>

  <!-- The name of the parameter under which the URDF is loaded -->
  <arg name="robot_description_name" default="robot_description"/>

  <!-- ik solver path -->
  <arg name="hand_ik_solver_path" default="$(find sr_moveit_hand_config)/config/kinematics_template.yaml"/>
  
  <!-- The name of the URDF file to load -->
  <arg name="robot_description" default="$(find sr_multi_description)/urdf/right_srhand_ur10.urdf.xacro"/>
  
  <!-- Load universal robot description format (URDF) -->
  <param if="$(arg load_robot_description)" name="$(arg robot_description_name)" command="xacro $(arg robot_description)"/>

  <!-- The name of the robot config file to load -->
  <arg name="robot_config_file" default=""/>

  <!-- The name of the config file to load saved move groups states from -->
  <arg name="move_group_states_file" default="$(find sr_multi_moveit_config)/config/multi_robot_move_group_states.yaml"/>

  <!-- The semantic description that corresponds to the URDF -->
  <node name="generate_robot_srdf" pkg="sr_multi_moveit_config" type="generate_robot_srdf.py" respawn="false" output="screen" args="$(arg robot_config_file) $(arg move_group_states_file)" />

  <!-- Load updated joint limits (override information from URDF) -->
  <group ns="$(arg robot_description_name)_planning">
    <node name="load_joint_limits" pkg="sr_multi_moveit_config" type="generate_load_moveit_config.py" respawn="false" output="screen" args="joint_limits $(arg robot_config_file)" />    
  </group>

  <!-- Load default settings for kinematics; these settings are overridden by settings in a node's namespace -->
  <group ns="$(arg robot_description_name)_kinematics">
    <node name="load_kinematics" pkg="sr_multi_moveit_config" type="generate_load_moveit_config.py" respawn="false" output="screen" args="kinematics $(arg robot_config_file) $(arg hand_ik_solver_path)" />
  </group>

</launch>
